thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -546,9 +565,32 @@ private void decodeAndInvoke(
             ctx.logBuilder().requestContent(null, null);
         }
 
+        if (!f.declaredDecorators().isEmpty()) {
+            ctx.setAttr(DECODED_REQUEST, new DecodedRequest(serializationFormat, seqId, f, decodedReq));
+            try {
+                httpRes.complete(decorate(f).serve(this, ctx, req.toHttpRequest()));
```

Comment:
I think we need to create the decorated THttpService once in `serviceAdded` instead of creating whenever it receives a request.
```java
dependencyInjector = cfg.server().config().dependencyInjector();
for (ThriftServiceEntry thriftServiceEntry : entries().values()) {
    for (ThriftFunction thriftFunction : thriftServiceEntry.metadata().functions().values()) {
        final List<DecoratorAndOrder> decorators = thriftFunction.declaredDecorators();
        if (!decorators.isEmpty()) {
            Function<? super HttpService, ? extends HttpService> decorator = Function.identity();
            for (int i = decorators.size() - 1; i >= 0; i--) {
                final DecoratorAndOrder d = decorators.get(i);
                decorator = decorator.andThen(d.decorator(dependencyInjector));
            }
            HttpService decoratedTHttpService = decorator.apply(this);
            thriftFunction.decoratedTHttpService(decoratedTHttpService);
        }
    }
}
```
then we can use it here.

Reply:
I tried to apply your comment! 
PTAL 🙇🏻